metabase/metabase
# fix charts crash in safari when dataset timezone is not in iana format

Closes https://github.com/metabase/metabase/issues/44128

### Description

The new implementation of cartesian charts use `dayjs` instead of `moment` since it has been deprecated. Dayjs relies on browser APIs but it can behave differently depending on a browser engine. Turns out Safari does not support non-IANA time zones, that are offsets, such as `+00:00` as they come from the backend side. Since there are cases when an arbitrary offset does not have a non-DST IANA timezone, I went with the approach of adding or subtracting the offset value in minutes.

### How to verify

- Revert all changes except for the new `safari-non-iana-timezone-repro-44128` story
- Run storybook
- Open the story in Safari (I could repro on 17.1.2)
- Ensure it fails
- Return back all the changes
- Ensure the story works in Safari

### Demo

Works in Safari:
<img width="801" alt="Screenshot 2024-06-14 at 3 01 05 PM" src="https://github.com/metabase/metabase/assets/14301985/83475292-4802-4da4-ab0c-4a333a466052">

### Checklist
